eslint for .jsx

Right now we don't do eslinting for .jsx.

One we do, we'll get following errors:

```
/Volumes/data/projects/meteor-samples/mantra-demo/client/components/layouts/main.jsx
   1:8   error  "Navigations" is defined but never used  no-unused-vars
   3:18  error  "content" is defined but never used      no-unused-vars
  11:6   error  "content23" is not defined               no-undef

✖ 3 problems (3 errors, 0 warnings)

Arunodas-MacBook:mantra-demo arunoda$ ./node_modules/.bin/eslint client/components/layouts/main
Arunodas-MacBook:mantra-demo arunoda$
Arunodas-MacBook:mantra-demo arunoda$
Arunodas-MacBook:mantra-demo arunoda$ npm run lint

> mantra-demo@0.0.0 lint /Volumes/data/projects/meteor-samples/mantra-demo
> eslint ./libs ./client ./server --ext .js --ext .jsx


/Volumes/data/projects/meteor-samples/mantra-demo/client/components/layouts/main.jsx
   1:8   error  "Navigations" is defined but never used  no-unused-vars

/Volumes/data/projects/meteor-samples/mantra-demo/client/components/newpost/index.jsx
  12:1  error  Line 12 exceeds the maximum line length of 80  max-len
  13:1  error  Line 13 exceeds the maximum line length of 80  max-len

/Volumes/data/projects/meteor-samples/mantra-demo/client/configs/routes.jsx
   1:23  error  "withContext" is defined but never used  no-unused-vars
   9:8   error  "PostList" is defined but never used     no-unused-vars
  10:8   error  "Post" is defined but never used         no-unused-vars
  11:8   error  "NewPost" is defined but never used      no-unused-vars

/Volumes/data/projects/meteor-samples/mantra-demo/client/libs/react-app-context.jsx
   6:3   error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in
   8:5   error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in
  14:20  error  "ChildComponent" is defined but never used                                                                  no-unused-vars
  43:20  error  "ChildComponent" is defined but never used                                                                  no-unused-vars

✖ 13 problems (13 errors, 0 warnings)
```

Somehow, it does not detect react component usage. That's there are `no-unsed-vars` issue.
Also we don't need to do 80 char limit for jsx. Do we? (That'd be weird then)

--- a/.eslintrc
+++ b/.eslintrc
@@ -5,6 +5,10 @@
     "node": true
   },
 
+  "plugins": [
+    "react"
+  ],
+
   "ecmaFeatures": {
     "arrowFunctions": true,
     "binaryLiterals": true,
@@ -65,7 +69,7 @@
     "linebreak-style": 2,
     "lines-around-comment": 0,
     "max-depth": 0,
-    "max-len": [2, 80, 4],
+    "max-len": [2, 100, 4],
     "max-nested-callbacks": 0,
     "max-params": 0,
     "max-statements": 0,

--- a/client/configs/routes.jsx
+++ b/client/configs/routes.jsx
@@ -1,4 +1,4 @@
-import {applyContext, withContext} from '../libs/react-app-context.jsx';
+import {applyContext} from '../libs/react-app-context.jsx';
 import context from './context';
 import actions from '../actions';
 

--- a/client/libs/react-app-context.jsx
+++ b/client/libs/react-app-context.jsx
@@ -3,12 +3,16 @@ import {React} from 'meteor/react-runtime';
 
 export function applyContext(context, _actions) {
   const actions = {};
-  for (var key in _actions) {
-    const actionMap = _actions[key];
-    for (var actionName in actionMap) {
-      actionMap[actionName] = actionMap[actionName].bind(null, context);
+  for (let key in _actions) {
+    if (_actions.hasOwnProperty(key)) {
+      const actionMap = _actions[key];
+      for (let actionName in actionMap) {
+        if (actionMap.hasOwnProperty(actionName)) {
+          actionMap[actionName] = actionMap[actionName].bind(null, context);
+        }
+      }
+      actions[key] = actionMap;
     }
-    actions[key] = actionMap;
   }
 
   return function (ChildComponent) {

--- a/package.json
+++ b/package.json
@@ -5,10 +5,11 @@
     "react-komposer": "1.0.0"
   },
   "devDependencies": {
-    "eslint": "1.10.x"
+    "eslint": "1.10.x",
+    "eslint-plugin-react": "3.14.x"
   },
   "scripts": {
-    "lint": "eslint ./libs ./client ./server"
+    "lint": "eslint ./libs ./client ./server --ext .js --ext .jsx"
   },
   "private": true
 }

